pkg/resources/resources_test.go

```diff
@@ -21,27 +21,28 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
-func createResources(cpu, memory int64) *Resources {
+func createResources(cpu, memory, nvidiaGpus int64) *Resources {
 	return &Resources{
-		CPU:    *resource.NewQuantity(cpu, resource.DecimalSI),
-		Memory: *resource.NewQuantity(memory, resource.BinarySI),
+		CPU:       *resource.NewQuantity(cpu, resource.DecimalSI),
+		Memory:    *resource.NewQuantity(memory, resource.BinarySI),
+		NvidiaGPU: *resource.NewQuantity(nvidiaGpus, resource.DecimalSI),
 	}
 }
 
 func TestAdd(t *testing.T) {
-	first := NodeGroupResources(map[string]*Resources{"1": createResources(1, 2), "2": createResources(3, 10)})
-	second := NodeGroupResources(map[string]*Resources{"1": createResources(2, 4), "3": createResources(1, 5)})
-	result := NodeGroupResources(map[string]*Resources{"1": createResources(3, 6), "2": createResources(3, 10), "3": createResources(1, 5)})
+	first := NodeGroupResources(map[string]*Resources{"1": createResources(1, 2, 3), "2": createResources(3, 10, 4)})
```

Comment:
For the sake of the practical similarity, could you add a few example tests in all methods with 0 GPUs?